source/Mlos.Python/mlos/Optimizers/OptimizerBase.py

```diff
@@ -40,13 +40,15 @@ def get_optimizer_convergence_state(self):
     def get_surrogate_model_fit_state(self):
         return self.get_optimizer_convergence_state().surrogate_model_fit_state
 
-    @abstractmethod
-    def suggest(self, random=False, context=None) -> Point:
-        """Suggest the next set of parameters to try.
+    @trace()
+    def suggest(self, context_values_dataframe=None):
+        """ Returns the next best configuration to try.
 
-        :return:
+        It does so by generating num_samples_per_iteration random configurations,
+        passing them through the utility function and selecting the configuration with
+        the highest utility value.
         """
-        raise NotImplementedError("All subclasses must implement this method.")
+
 
     @abstractmethod
     def register(self, feature_values_pandas_frame, target_values_pandas_frame) -> None:

```

Comment:
Would it make sense to split up the feature_values into: parameters_df and context_df? This way the user does not to do the prefixing...